File: Module_Base_De_Donnees/Tp 2eme annee/Tp NoSQL/Tp 3/tp3.js
use tp2;
db.Film.insertOne({
 "title": "Inception",
 "genre": "Science Fiction",
 "releaseYear": 2010,
 "director": "Christopher Nolan",
 "actors": ["Leonardo DiCaprio", "Ellen Page", "Tom Hardy"],
 "rating": 8.7
})
db.Film.find()

//Q 1
db.Film.updateOne(
 { _id: ObjectId("658bf7f393764ecea4912b2e") },
 { $inc: { rating: 0.3 } }
);

//Q 2
db.Film.deleteMany({"releaseYear": {"$lt": 2000}});

//Q 3
db.Film.updateOne(
{"_id": ObjectId("658bf7f393764ecea4912b2e")},
{$set: {"budjet": "100M $"}}
);

//Q 4
db.Film.updateMany(
{"director": "Christopher Nolan"},
{$push: {"actors": "Cillian Murphy"}}
);

//Q 5
db.Film.deleteMany({"rating": {"$lt": 5}});

//Q 6
db.Film.updateMany(
{"genre": "Science Fiction"},
{$set: {"releaseYear": 2015}}
);

//Q 7
db.Film.updateMany(
{"genre": "Science Fiction", "releaseYear": {"$gt": 2010}},
{$inc: {"rating": 0.5}}
);

//Q 8
db.Film.deleteMany({"genre": {"$nin": ["Science Fiction", "Action"]}});

//Q 9
db.Film.updateMany(
{},
{$addToSet: {"reviews": []}}
);

//Q 10
function getDate() {
  let date_sys = new Date().getFullYear();
  let date = [];
};
db.Film.updateMany(
{},
{$set: {"age": new Date().getFullYear() - "$releaseYear"}}
);

//Q 11
db.Film.updateMany(
{},
{$pull: {"actors": []}}
);

//Q 12
db.Film.updateMany(
{"releaseYear": {"$gt": 2010}},
{$set: {"title": {$concat: ["$title", "3D"]}}}
);

File: Module_Base_De_Donnees/Tp 2eme annee/Tp NoSQL/Tp pyMongo/Tp 3/tp3.py
from pymongo import MongoClient
from datetime import datetime

conx = MongoClient("mongodb://localhost:27017/")

db = conx["tp2"]

# A
Film = db["Film"]

Film.insert_one(
    {
        "title": "Inception",
        "genre": "Science Fiction",
        "releaseYear": 2010,
        "director": "Christopher Nolan",
        "actors": ["Leonardo DiCaprio", "Ellen Page", "Tom Hardy"],
        "rating": 8.7
    },
)

# 1
Film.update_one({"title": "Inception"}, {"$inc": {"rating": 0.3}})

# 2
Film.delete_many({"releaseYear": {"$lt": 2000}})

# 3
Film.update_one({"title": "Inception"}, {"$set": {"budget": "100M$"}})

# 4
Film.update_many({"director": "Christopher Nolan"}, {"$push": {"actors": "Cillian Murphy"}})

# 5
Film.delete_many({"rating": {"$lt": 5.0}})

# 6
Film.update_many({"genre": "Science Fiction"}, {"$set": {"releaseYear": 2015}})

# 7
Film.update_many({"$and": [{"genre": "Science Fiction"}, {"releaseYear": {"$gt": 2010}}]}, {"$inc": {"rating": 0.5}})

# 8
Film.delete_many({"genre": {"$nin": ["Science Fiction", "Action"]}})

# 9
Film.update_many({"reviews": {"$exists": False}}, {"$set": {"reviews": []}})

# 10
Film.update_many({}, {"$set": {"age": {"$subtract": [datetime.now(), "$releaseYear"]}}})

# 11
Film.delete_many({"actors": []})

# 12
Film.update_many({"releaseYear": {"$gt": 2010}}, {"$set": {"titre": {"$concat": ["$titre", " 3D"]}}})